File: backend/src/001-matematicas/algebra_aritmetica/001-basica/aritmetica.go
package main

import (
   "fmt"
   "math"
)

func main() {
    // Las funciones matemáticas del paquete 'math'
    // sólo aceptan valores de tipo 'float64'.
    num := 50.4

    // Raíz cuadrada
    raiz_cuadrada := math.Sqrt(num)
    fmt.Printf("%v\n", raiz_cuadrada)

    // Próximo entero hacia arriba
    prox_int_up := math.Ceil(num)
    fmt.Printf("%v\n", prox_int_up)

    // Próximo entero hacia abajo
    prox_int_down := math.Floor(num)
    fmt.Printf("%v\n", prox_int_down)

    // Valor absoluto
    abs_value := math.Abs(num)
    fmt.Printf("%v\n", abs_value)

    // Cálculo de factorial
    // https://gist.github.com/esimov/9622710

    // Logaritmo natural
    log := math.Log(num)
    fmt.Printf("%v\n", log)

    // Logaritmo en base 10
    log10 := math.Log10(num)
    fmt.Printf("%v\n", log10)
}

/* Fuentes:
https://golang.org/pkg/math/#Abs
https://gist.github.com/esimov/9622710
*/

File: backend/src/001-matematicas/algebra_aritmetica/001-basica/aritmetica.c
#include <stdio.h>
#include <math.h>

/* Para linkear la biblioteca <math.h> usamos la opción '-lm'
$ gcc aritmetica.c -o aritmetica -lm
*/

int main() {
    float num, calc;
    printf("Introduce un número: ");
    scanf("%f", &num);
    printf("\n");

    // Raíz cuadrada
    if (num > 0) {
        calc = sqrt(num);
        printf("\tRaíz cuadrada = %.2f\n", calc);
    }

    // Próximo entero hacia arriba
    calc = ceil(num);
    printf("\tPróximo entero hacia arriba = %.2f\n", calc);

    // Próximo entero hacia abajo
    calc = floor(num);
    printf("\tPróximo entero hacia abajo = %.2f\n", calc);

    // Valor absoluto
    calc = fabs(num);
    printf("\tValor absoluto = %.2f\n", calc);

    // Logaritmo natural
    calc = log(num);
    printf("\tLogaritmo natural = %.2f\n", calc);

    // Logaritmo en base 10
    calc = log10(num);
    printf("\tLogaritmo en base 10 = %.2f\n", calc);

    return 0;
}

/* Fuentes:
https://es.wikipedia.org/wiki/Math.h
*/
